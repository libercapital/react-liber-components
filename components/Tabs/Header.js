'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Header = function Header(_ref) {
  var children = _ref.children,
      labels = _ref.labels,
      slider = _ref.slider,
      header = _ref.header,
      asList = _ref.asList;
  return _react2.default.createElement(
    _react.Fragment,
    null,
    asList ? children(labels, slider, header) : children(labels)
  );
};

Header.propTypes = {
  children: _propTypes2.default.func,
  labels: _propTypes2.default.oneOfType([_propTypes2.default.any]),
  slider: _propTypes2.default.oneOfType([_propTypes2.default.any]),
  header: _propTypes2.default.oneOfType([_propTypes2.default.any]),
  asList: _propTypes2.default.bool
};

Header.defaultProps = {
  children: function children() {},
  labels: null,
  slider: null,
  header: null,
  asList: false
};

exports.default = Header;