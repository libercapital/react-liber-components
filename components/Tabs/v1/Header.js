"use strict";

require("core-js/modules/es6.object.define-property");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Header = function Header(_ref) {
  var children = _ref.children,
      labels = _ref.labels,
      slider = _ref.slider,
      header = _ref.header,
      asList = _ref.asList;
  return _react["default"].createElement(_react.Fragment, null, asList ? children(labels, slider, header) : children(labels));
};

Header.propTypes = {
  children: _propTypes["default"].func,
  labels: _propTypes["default"].oneOfType([_propTypes["default"].any]),
  slider: _propTypes["default"].oneOfType([_propTypes["default"].any]),
  header: _propTypes["default"].oneOfType([_propTypes["default"].any]),
  asList: _propTypes["default"].bool
};
Header.defaultProps = {
  children: function children() {},
  labels: null,
  slider: null,
  header: null,
  asList: false
};
var _default = Header;
exports["default"] = _default;