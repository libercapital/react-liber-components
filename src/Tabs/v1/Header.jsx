import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Header = ({
  children, labels, slider, header, asList,
}) => (
  <Fragment>
    {
      asList
        ? children(labels, slider, header)
        : children(labels)
    }
  </Fragment>
);

Header.propTypes = {
  children: PropTypes.func,
  labels: PropTypes.oneOfType([PropTypes.any]),
  slider: PropTypes.oneOfType([PropTypes.any]),
  header: PropTypes.oneOfType([PropTypes.any]),
  asList: PropTypes.bool,
};

Header.defaultProps = {
  children: () => {},
  labels: null,
  slider: null,
  header: null,
  asList: false,
};

export default Header;
