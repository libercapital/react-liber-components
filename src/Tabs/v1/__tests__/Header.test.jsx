import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import Header from '../Header';

describe('Testing Header Component', () => {
  let props;
  configure({ adapter: new Adapter() });

  beforeEach(() => {
    props = {
      children: sinon.spy(),
      labels: 'labels',
    };
  });
  it('should render correctly empty', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly empty', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(props.children.calledWith(props.labels)).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
