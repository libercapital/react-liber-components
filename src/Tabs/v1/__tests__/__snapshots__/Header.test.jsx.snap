// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component should render correctly empty 1`] = `<Fragment />`;

exports[`Testing Header Component should render correctly empty 2`] = `<Fragment />`;
